src/hooks/UseSignUpForm.jsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { register } from '../api/DBRequests';
+import { useAuth } from '../context/AuthProvider';
+
+const UseSignUpForm = () => {
```

Comment:
As this is a React custom hook, I think you should rename both the function and the filename to `useSignUpForm`